### PR TITLE
Fix clown shoes in pubsub validation

### PR DESCRIPTION
**Pull request by vyzo**
We should always accept our own self-published messages, without lengthy validation for blocks and re-adding to the mpool for messages.
Edit: we actually need to do some lightweight validation for sanity checks, there is that.

Clown shoes were size ~2^31-1~ 2^16-1.

### CODE REVIEW

File: chain/sub/incoming.go [hunk 1]
```diff
@@ -332,6 +339,42 @@ func (bv *BlockValidator) Validate(ctx context.Context, pid peer.ID, msg *pubsub
 	return pubsub.ValidationAccept
 }
 
+func (bv *BlockValidator) validateLocalBlock(ctx context.Context, msg *pubsub.Message) pubsub.ValidationResult {
```
Comment (whyrusleeping) [inline chain/sub/incoming.go:342]: maybe we can dedupe some of this logic?

Comment (vyzo) [reply chain/sub/incoming.go:342]: hrm yes, we probably can :)

Comment (vyzo) [reply chain/sub/incoming.go:327]: done, deduplicated the common code.

**Comment by vyzo** [comment]
linter is complaining for irrelevant stuff...

**Pull request merged**